lib/rex/post/smb/ui/console/command_dispatcher/shares.rb

```diff
@@ -318,7 +344,198 @@ def cmd_cat(*args)
           def cmd_cd_tabs(_str, words)
             return [] if words.length > 1
 
-            @@cat_opts.option_keys
+            @@cd_opts.option_keys
+          end
+
+          def cmd_upload(*args)
+            if args.include?('-h') || args.include?('--help')
+              cmd_upload_help
+              return
+            end
+            local_path = nil
+            remote_path = nil
+
+            @@upload_opts.parse(args) do |_opt, idx, val|
+              case idx
+              when 0
+                local_path = val
+              when 1
+                remote_path = val
+              else
+                print_warning('Too many parameters')
+                cmd_upload_help
+              end
+            end
+
+            if local_path.blank?
+              print_error('No local path given')
+              return
+            end
+
+            remote_path = Rex::Post::File.basename(local_path) if remote_path.nil?
+            full_path = as_ntpath(Pathname.new(shell.cwd).join(remote_path).to_s)
+
+            fd = simple_client.open(full_path, 'wct', write: true)
+
+            data = ::File.read(local_path, ::File.size(local_path), mode: 'rb')
+            fd.write(data)
+            fd.close
+
+            print_good("#{local_path} uploaded to #{full_path}")
+          end
+
+          def cmd_upload_tabs(str, words)
+            tab_complete_filenames(str, words)
+          end
+
+          def cmd_upload_help
+            print_line 'Usage: upload <local_path> <remote_path>'
+            print_line
+            print_line 'Upload a file to the remote target.'
+            print @@upload_opts.usage
+          end
+
+          def cmd_download(*args)
+            if args.include?('-h') || args.include?('--help')
+              cmd_download_help
+              return
+            end
+            remote_path = nil
+            local_path = nil
+
+            @@download_opts.parse(args) do |_opt, idx, val|
+              case idx
+              when 0
+                remote_path = val
+              when 1
+                local_path = val
+              else
+                print_warning('Too many parameters')
+                cmd_download_help
+              end
+            end
+
+            if remote_path.blank?
+              print_error('No remote path given')
+              return
+            end
+
+            local_path = Rex::Post::File.basename(remote_path) if local_path.nil?
+            full_path = as_ntpath(Pathname.new(shell.cwd).join(remote_path).to_s)
+
+            fd = simple_client.open(full_path, 'o')
+            data = fd.read
+            fd.close
+
+            ::File.open(local_path, 'wb') { |local_file| local_file.write(data) }
```

Comment:
```suggestion
            ::File.binwrite(local_path, data)
```

Assuming `data` is raw bytes